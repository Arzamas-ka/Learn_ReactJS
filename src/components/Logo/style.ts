import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { colors } from '../../styles/consts';

export const StyledLogo = styled(Link)`
  padding: 25px 55px;
  align-self: baseline;

  color: ${colors.coral};
  text-decoration: none;

  :hover {
    color: ${colors.white};
  }
`;

export const StyledSpan = styled.span`
  font-weight: 400;
`;
