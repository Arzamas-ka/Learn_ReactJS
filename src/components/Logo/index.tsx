import * as React from 'react';
import { FC } from 'react';

import { StyledLogo, StyledSpan } from './style';

const Logo: FC = () => (
  <StyledLogo to="/">
    <h2>
      netflix<StyledSpan>roulette</StyledSpan>
    </h2>
  </StyledLogo>
);

export default Logo;
