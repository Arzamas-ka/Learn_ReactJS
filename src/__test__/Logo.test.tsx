import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';

import Logo from '../components/Logo/index';

it('<Logo />', () => {
  render(
    <Router>
      <Logo />
    </Router>,
  );
  const logoText = screen.getByText('netflix');

  expect(logoText).toBeInTheDocument();
});
