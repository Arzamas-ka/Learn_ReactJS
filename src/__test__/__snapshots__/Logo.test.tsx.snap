// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> 1`] = `
<body>
  <div>
    <a
      class="sc-bdfBwQ bfKVhG"
      href="/"
    >
      <h2>
        netflix
        <span
          class="sc-gsTCUz fYWTln"
        >
          roulette
        </span>
      </h2>
    </a>
  </div>
</body>
`;

exports[`<Logo /> matches with snapshot 1`] = `
<body>
  <div>
    <a
      class="sc-bdfBwQ bfKVhG"
      href="/"
    >
      <h2>
        netflix
        <span
          class="sc-gsTCUz fYWTln"
        >
          roulette
        </span>
      </h2>
    </a>
  </div>
</body>
`;

exports[`<Logo /> shows text 1`] = `
<body>
  <div>
    <a
      class="sc-bdfBwQ bfKVhG"
      href="/"
    >
      <h2>
        netflix
        <span
          class="sc-gsTCUz fYWTln"
        >
          roulette
        </span>
      </h2>
    </a>
  </div>
</body>
`;
